packages/accounts-controller/src/AccountsController.test.ts

```diff
@@ -1614,32 +1736,260 @@ describe('AccountsController', () => {
+      id: 'mock-id-2',
+      name: 'mock account 2',
+      address: 'mock-address-2',
+      keyringType: KeyringTypes.hd,
+      lastSelected: 22222,
+    });
+
+    it.each([
+      {
+        lastSelectedAccount: mockNewerEvmAccount,
+        expected: mockNewerEvmAccount,
+      },
+      {
+        lastSelectedAccount: mockOlderEvmAccount,
+        expected: mockOlderEvmAccount,
+      },
+      {
+        lastSelectedAccount: mockNonEvmAccount,
+        expected: mockNewerEvmAccount,
+      },
+    ])(
+      'last selected account type is $lastSelectedAccount.type should return the selectedAccount with id $expected.id',
+      ({ lastSelectedAccount, expected }) => {
+        const { accountsController } = setupAccountsController({
+          initialState: {
+            internalAccounts: {
+              accounts: {
+                [mockOlderEvmAccount.id]: mockOlderEvmAccount,
+                [mockNewerEvmAccount.id]: mockNewerEvmAccount,
+                [mockNonEvmAccount.id]: mockNonEvmAccount,
+              },
+              selectedAccount: lastSelectedAccount.id,
+            },
+          },
+        });
+
+        expect(accountsController.getSelectedAccount()).toStrictEqual(expected);
+      },
+    );
+
+    it("should throw error if there aren't any evm accounts", () => {
+      const { accountsController } = setupAccountsController({
+        initialState: {
+          internalAccounts: {
+            accounts: {
+              [mockNonEvmAccount.id]: mockNonEvmAccount,
+            },
+            selectedAccount: mockNonEvmAccount.id,
+          },
+        },
+      });
+
+      expect(() => accountsController.getSelectedAccount()).toThrow(
+        'AccountsController: No EVM accounts',
+      );
+    });
+  });
+
+  describe('getSelectedMultichainAccount', () => {
+    const mockNonEvmAccount = createExpectedInternalAccount({
+      id: 'mock-non-evm',
+      name: 'non-evm',
+      address: 'bc1qzqc2aqlw8nwa0a05ehjkk7dgt8308ac7kzw9a6',
+      keyringType: KeyringTypes.snap,
+      type: BtcAccountType.P2wpkh,
+    });
+
+    const mockOlderEvmAccount = createExpectedInternalAccount({
+      id: 'mock-id-1',
+      name: 'mock account 1',
+      address: 'mock-address-1',
+      keyringType: KeyringTypes.hd,
+      lastSelected: 11111,
+    });
+    const mockNewerEvmAccount = createExpectedInternalAccount({
+      id: 'mock-id-2',
+      name: 'mock account 2',
+      address: 'mock-address-2',
+      keyringType: KeyringTypes.hd,
+      lastSelected: 22222,
+    });
+
+    it.each([
+      {
+        chainId: undefined,
+        selectedAccount: mockNewerEvmAccount,
+        expected: mockNewerEvmAccount,
+      },
+      {
+        chainId: undefined,
+        selectedAccount: mockNonEvmAccount,
+        expected: mockNonEvmAccount,
+      },
+      {
+        chainId: 'eip155:1',
+        selectedAccount: mockNonEvmAccount,
+        expected: mockNewerEvmAccount,
+      },
+      {
+        chainId: 'bip122:1',
```

Comment:
Just for consistency:

```suggestion
        chainId: 'bip122:000000000019d6689c085ae165831e93',
```